sources/nodes/kube.go

```diff
@@ -78,8 +78,15 @@ func (self *kubeNodes) List() (*NodeList, error) {
 			nodeList.Items[Host(node.Name)] = Info{PublicIP: node.Status.HostIP, InternalIP: addrs[0].String()}
 			goodNodes = append(goodNodes, node.Name)
 		} else {
-			glog.Errorf("Skipping host %s since looking up its IP failed - %s", node.Name, err)
-			self.recordNodeError(node.Name)
+			ip := net.ParseIP(node.Name)
```

Comment:
We can, some systems use IP instead to avoid hostname resolution issues.

Le Mon Feb 16 2015 at 19:03:03, Vish Kannan <notifications@github.com> a
écrit :

> In sources/nodes/kube.go
> <https://github.com/GoogleCloudPlatform/heapster/pull/105#discussion_r24766205>
> :
>
> > @@ -78,8 +78,15 @@ func (self *kubeNodes) List() (*NodeList, error) {
> >  			nodeList.Items[Host(node.Name)] = Info{PublicIP: node.Status.HostIP, InternalIP: addrs[0].String()}
> >  			goodNodes = append(goodNodes, node.Name)
> >  		} else {
> > -			glog.Errorf("Skipping host %s since looking up its IP failed - %s", node.Name, err)
> > -			self.recordNodeError(node.Name)
> > +			ip := net.ParseIP(node.Name)
>
> Is this a custom cluster setup where the node name is its IP? Why so? Why
> not give the node an unique hostname that can be resolves to an IP?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/heapster/pull/105/files#r24766205>
> .
>